### ISSUE TITLE
[Feature] support a different path for the production manifest

### ISSUE THREAD

**Issue by noel** [DataRecce/recce#434]
### Describe the feature

Today Recce expects the production manifest to be at `target-base/` I would like to pass a different location for the manifest.  Ideally there would be a parameter like `--base-manifest-path` where I can pass a different location.

**Comment by popcornylu** [comment]
Thank you for your suggestion. 

In dbt, a similar option is `--target-path`, so we may lean towards using `--target-base-path` for consistency with other dbt commands. Additionally, this path doesn't just include the manifest, but also other artifacts like the catalog.

**Comment by noel** [comment]
Actually Dbt flag for this is —state

**Comment by popcornylu** [comment]
Yes, thanks for the reminder. Indeed, the closest equivalent in dbt is `--state`, and we do use a mechanism like `-s state:modified` to determine modified nodes.

However, in the context of Recce, `--state` might conflict with specifying a state file. We are considering options like `--state`, `--dbt-state`, and `--target-base-path` to choose the most suitable one.

**Comment by popcornylu** [comment]
Decide to use `--target-base-path`, and we will implement `--target-path` as well.

**Comment by noel** [comment]
Great. Thanks On Oct 1, 2024, at 7:07 PM, popcornylu ***@***.***> wrote:﻿
Decide to use --target-base-path, and we will implement --target-path as well.
